lib/internal/util.js

```diff
@@ -187,3 +189,36 @@ exports.convertToValidSignal = function convertToValidSignal(signal) {
 
   throw new Error('Unknown signal: ' + signal);
 };
+
+const kCustomPromisifiedSymbol = Symbol('util.promisify.custom');
+function promisify(orig) {
+  if (typeof orig !== 'function') {
+    throw new TypeError('The first argument to promisify() must be a function');
+  }
+
+  if (orig[kCustomPromisifiedSymbol]) {
+    return orig[kCustomPromisifiedSymbol];
+  }
+
+  const fn = function(...args) {
+    const promise = createPromise();
+    try {
+      orig.call(this, ...args, (err, data) => {
+        if (err) {
+          promiseReject(promise, err);
+        } else {
+          promiseResolve(promise, data);
+        }
+      });
+    } catch (err) {
+      promiseReject(promise, err);
+    }
+    return promise;
+  };
+
+  return Object.defineProperties(fn, Object.getOwnPropertyDescriptors(orig));
```

Comment:
@TimothyGu yeah … that thought occurred to me, but I basically didn’t see how setting it would make sense here. If you prefer, I’ll add

```js
Object.setPrototypeOf(fn, Object.getPrototypeOf(orig));
```